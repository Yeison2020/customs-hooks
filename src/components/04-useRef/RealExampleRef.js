import React, { useState, useRef } from "react";
import "../02-useEffect/effects.css";
import MultipleCustomsHooks from "../03-examples/MultipleCustomsHooks";

const RealExampleRef = () => {
  // This Example hide and show my MultipleCustomsHooks Still not use of the useRef
  const [show, setShow] = useState(false);
  return (
    <div>
      <h1>Real Ref</h1>
      <hr />
      {show && <MultipleCustomsHooks />}
      <button
        className="btn btn-primary mt-5"
        onClick={() => {
          setShow(!show);
        }}
      >
        Show/Hide
      </button>
    </div>
  );
};

export default RealExampleRef;
