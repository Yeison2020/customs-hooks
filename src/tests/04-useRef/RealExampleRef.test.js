import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import RealExampleRef from "../../components/04-useRef/RealExampleRef";
import { renderHook, act } from "@testing-library/react-hooks";

describe("Pruebas en <RalExampleUseRef/>", () => {
  test("Should Macth with the snapShot", () => {
    const wrapper = <RealExampleRef />;
    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.find("MultipleCustomsHooks").exist()).toBe(false);
  });

  test("Should show the component when called functions", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomsHooks").exists()).toBe(false);
  });
});
