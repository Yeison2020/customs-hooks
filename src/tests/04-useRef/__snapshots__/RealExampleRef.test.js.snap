// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RalExampleUseRef/> Should Macth with the snapShot 1`] = `<RealExampleRef />`;
